ptvsd/_vendored/pydevd/_pydevd_bundle/pydevd_io.py

```diff
@@ -6,25 +6,29 @@ class IORedirector:
     '''This class works to redirect the write function to many streams
     '''
 
-    def __init__(self, *args):
-        self._redirectTo = args
+    def __init__(self, original, new_redirect, wrap_buffer=False):
+        self._redirect_to = (original, new_redirect)
+        if wrap_buffer:
+            self.buffer = IORedirector(original.buffer, new_redirect.buffer, False)
 
     def write(self, s):
-        for r in self._redirectTo:
-            try:
-                r.write(s)
-            except:
-                pass
+        # The original one may fail if the user tries to write a number to the stream, etc.
+        self._redirect_to[0].write(s)
+        try:
+            # This one should not fail.
```

Comment:
If this should not fail, the why is it in `try`?
Should the comment read `This should be allowed to fail`?

Reply:
The comment should actually be that it is not allowed to fail, but thinking about it a bit more I don't think that the except should be needed at all (if it fails it's probably an error and we want to fix it), so, removed that try..except.